pkg/flare/archive.go

```diff
@@ -726,9 +734,13 @@ func walkConfigFilePaths(tempDir, hostname string, confSearchPaths SearchPaths,
 					return err
 				}
 
-				if permsInfos != nil {
+				if len(permsInfos) != 0 {
 					permsInfos.add(src)
-					addParentPerms(filePath, permsInfos)
+					absPath, err := filepath.Abs(filePath)
+					if err != nil {
+						log.Errorf("Error while getting absolute file path for parent directory: %v", err)
+					}
+					addParentPerms(absPath, permsInfos)
```

Comment:
Note: If `addParentPerms` assumes the path is absolute, what do you think about moving `absPath, err := filepath.Abs(filePath)` inside `addParentPerms`